Client/Frontend/Popup/AlertPopupView.swift

```diff
@@ -56,32 +56,54 @@ class AlertPopupView: PopupView {
     }
     
     func updateSubviews() {
+        titleLabel.adjustsFontSizeToFitWidth = false
+        messageLabel.adjustsFontSizeToFitWidth = false
+
+        _updateSubviews()
+        
+        let paddingHeight: CGFloat = padding * 3.0
+        let externalContentHeight: CGFloat = dialogButtons.count == 0 ? paddingHeight
+            : kPopupDialogButtonHeight + paddingHeight
+        
+        if containerView.frame.height + externalContentHeight > UIScreen.main.bounds.height {
+            let desiredHeight: CGFloat = UIScreen.main.bounds.height - externalContentHeight
+            let resizePercentage = desiredHeight / containerView.frame.height
+            titleLabel.adjustsFontSizeToFitWidth = true
+            messageLabel.adjustsFontSizeToFitWidth = true
+            _updateSubviews(resizePercentage)
+        }
+    }
+    
+    fileprivate func _updateSubviews(_ resizePercentage: CGFloat = 1.0) {
         let width: CGFloat = dialogWidth
         
         var imageFrame: CGRect = dialogImage?.frame ?? CGRect.zero
         if let dialogImage = dialogImage {
+            imageFrame.size = CGSize(width: dialogImage.image!.size.width * resizePercentage, height: dialogImage.image!.size.height * resizePercentage)
```

Comment:
Remove `image` forced casting (can unwrap inside the above if statement:

`if let dialogImage = dialogImage, let dialogeImageSize = dialogImage.image?.size`